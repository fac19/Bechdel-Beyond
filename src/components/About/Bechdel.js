import React from 'react';
import { makeStyles, useTheme } from '@material-ui/core/styles';
import MobileStepper from '@material-ui/core/MobileStepper';
import KeyboardArrowLeft from '@material-ui/icons/KeyboardArrowLeft';
import KeyboardArrowRight from '@material-ui/icons/KeyboardArrowRight';
import Typography from '@material-ui/core/Typography';
import Button from '@material-ui/core/Button';
import Paper from '@material-ui/core/Paper';
import Link from '@material-ui/core/Link';
import { Link as RouterLink } from 'react-router-dom';
import Divider from '@material-ui/core/Divider';
import FormControlLabel from '@material-ui/core/FormControlLabel';
import RadioGroup from '@material-ui/core/RadioGroup';
import Radio from '@material-ui/core/Radio';

const useStyles = makeStyles((theme) => ({
	root: {
		marginTop: '3rem',
		maxWidth: 400,
		flexGrow: 1,
		margin: '0 auto',
	},
	title: {
		marginTop: '3rem',
		textAlign: 'center',
		fontSize: '2rem',
		fontFamily: 'Merriweather, serif',
		fontWeight: 'bold',
	},
	subtitle: {
		marginTop: '1rem',
		textAlign: 'center',
		fontSize: '0.7rem',
		fontFamily: 'Merriweather, serif',
	},
	stepperLine: {
		width: '70%',
		margin: '0 auto',
	},
	button: {
		display: 'block',
		maxWidth: '40%',
		margin: '0 auto',
		marginTop: '4rem',
	},
	buttonLink: {
		fontSize: '0.7rem',
		fontFamily: 'Montserrat, sans-serif',
	},
	beyond: {
		textAlign: 'center',
	},
	beyondHeader: {
		marginTop: '3rem',
		fontSize: '1.5rem',
		fontFamily: 'Merriweather, serif',
	},
	questions: {
		marginTop: '3rem',
		fontFamily: 'Montserrat, sans-serif',
	},
	demos: {
		display: 'flex',
		justifyContent: 'center',
		padding: '5px 7px',
	},
	paper: {
		margin: '4% 10%',
		padding: theme.spacing(2),
		textAlign: 'center',
		color: theme.palette.text.secondary,
	},
	emojiBox: {
		margin: '2% auto',
		display: 'block',
		maxWidth: '50%',
		fontSize: '1.5rem',
	},
}));

export default function BechdelPage() {
	const classes = useStyles();
	const theme = useTheme();
	const [activeStep, setActiveStep] = React.useState(0);

	const handleNext = () => {
		setActiveStep((prevActiveStep) => prevActiveStep + 1);
	};

	const handleBack = () => {
		setActiveStep((prevActiveStep) => prevActiveStep - 1);
	};

	return (
		<>
			{activeStep === 0 ? (
				<>
					<Typography className={classes.title} variant="h3">
						The Bechdel Test
					</Typography>
					<div className={classes.subtitle}>
						<Link to="">Alison Bechdel , Dykes to Watch Out For, 1985</Link>
					</div>

					<Paper className={classes.paper}>
						A way of evaluating whether or not a movie portrays women in a way
						that is sexist or characterized by gender stereotyping
					</Paper>
				</>
			) : activeStep === 1 ? (
				<>
					<Typography className={classes.title} variant="h4">
						A Movie passes if...
					</Typography>
					<Paper className={classes.paper}>There are at least 2 women</Paper>
					<Paper className={classes.paper}>That talk to each other</Paper>
					<Paper className={classes.paper}>
						About something other than men.
					</Paper>
				</>
			) : (
				<div className={classes.beyond}>
					<Typography className={classes.beyondHeader} variant="h6">
						Help others find more diverse and interesting movies by sharing:
					</Typography>

					<Typography className={classes.questions}>
						Does the movie pass the Bechdel Test?
					</Typography>
					<RadioGroup className={classes.demos} name="Test" row>
						<FormControlLabel value="true" label="Yes" control={<Radio />} />
						<FormControlLabel value="false" label="No" control={<Radio />} />
					</RadioGroup>
					<Typography className={classes.questions}>
						Are you satisfied with the gender representation?
					</Typography>
					<div className={classes.emojiBox}>
						<span role="img" aria-label="emojis" className={classes.demos}>
							ðŸ¤¬ ðŸ˜ž ðŸ˜• ðŸ™‚ ðŸ’–
						</span>
					</div>
				</div>
			)}
			<Divider className={classes.stepperLine} />

			<MobileStepper
				variant="dots"
				steps={3}
				position="static"
				activeStep={activeStep}
				className={classes.root}
				nextButton={
					<Button size="small" onClick={handleNext} disabled={activeStep === 2}>
						Next
						{theme.direction === 'rtl' ? (
							<KeyboardArrowLeft />
						) : (
							<KeyboardArrowRight />
						)}
					</Button>
				}
				backButton={
					<Button size="small" onClick={handleBack} disabled={activeStep === 0}>
						{theme.direction === 'rtl' ? (
							<KeyboardArrowRight />
						) : (
							<KeyboardArrowLeft />
						)}
						Back
					</Button>
				}
			/>
			<Button className={classes.button} variant="outlined">
				{activeStep !== 2 ? (
					<Link className={classes.buttonLink} component={RouterLink} to="/">
						Skip to the movies
					</Link>
				) : (
					<Link className={classes.buttonLink} component={RouterLink} to="/">
						Show me the movies!
					</Link>
				)}
			</Button>
		</>
	);
}
