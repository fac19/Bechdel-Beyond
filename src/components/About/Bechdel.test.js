import React from 'react';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';

import Bechdel from './Bechdel';

it('should render Bechdel page correctly', () => {
	const content = renderer
		.create(
			<Router>
				<Bechdel />
			</Router>,
		)
		.toJSON();
	expect(content).toMatchSnapshot();
});
