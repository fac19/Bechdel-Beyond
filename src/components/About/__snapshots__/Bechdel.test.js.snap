// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Bechdel page correctly 1`] = `
Array [
  <h3
    className="MuiTypography-root makeStyles-title-2 MuiTypography-h3"
    data-cy="bechdel-setp-one"
  >
    The Bechdel Test
  </h3>,
  <div
    className="makeStyles-subtitle-3"
    data-cy="bechdel-author"
  >
    <a
      className="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      onBlur={[Function]}
      onFocus={[Function]}
      to=""
    >
      Alison Bechdel , Dykes to Watch Out For, 1985
    </a>
  </div>,
  <div
    className="MuiPaper-root makeStyles-paper-11 MuiPaper-elevation1 MuiPaper-rounded"
    data-cy="bechdel-idea"
  >
    A way of evaluating whether or not a movie portrays women in a way that is sexist or characterized by gender stereotyping
  </div>,
  <hr
    className="MuiDivider-root makeStyles-stepperLine-4"
  />,
  <div
    className="MuiPaper-root MuiMobileStepper-root MuiMobileStepper-positionStatic makeStyles-root-1 MuiPaper-elevation0"
  >
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall Mui-disabled Mui-disabled"
      data-cy="bechdel-back"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={-1}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
          />
        </svg>
        Back
      </span>
    </button>
    <div
      className="MuiMobileStepper-dots"
    >
      <div
        className="MuiMobileStepper-dot MuiMobileStepper-dotActive"
      />
      <div
        className="MuiMobileStepper-dot"
      />
      <div
        className="MuiMobileStepper-dot"
      />
    </div>
    <button
      className="MuiButtonBase-root MuiButton-root MuiButton-text MuiButton-textSizeSmall MuiButton-sizeSmall"
      data-cy="bechdel-next"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiButton-label"
      >
        Next
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M8.59 16.59L13.17 12 8.59 7.41 10 6l6 6-6 6-1.41-1.41z"
          />
        </svg>
      </span>
    </button>
  </div>,
  <button
    className="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-button-5"
    disabled={false}
    onBlur={[Function]}
    onDragLeave={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex={0}
    type="button"
  >
    <span
      className="MuiButton-label"
    >
      <a
        className="MuiTypography-root MuiLink-root MuiLink-underlineHover makeStyles-buttonLink-6 MuiTypography-colorPrimary"
        href="/"
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
      >
        Skip to the movies
      </a>
    </span>
  </button>,
]
`;
